### PR TITLE
XW-2090 | use icons from DS + changes in loading svgs

### PR DESCRIPTION
**Pull request by latata**
http://wikia-inc.atlassian.net/browse/XW-2090

### CODE REVIEW

File: app/components/curated-content-item.js [hunk 1]
```diff
@@ -37,21 +37,28 @@ export default Component.extend(
...
+					blog: 'clock'
+				};
 
 			let iconType;
 
-			if (typesWithDedicatedIcon.indexOf(type) > -1) {
-				iconType = type;
+			if (typesWithDedicatedIcon[type]) {
+				iconType = typesWithDedicatedIcon[type];
 			} else if (type === 'section') {
```
Comment (rybmat) [inline app/components/curated-content-item.js]: can we add `section:'grid'` to `typesWithDedicatedIcon` instead of this `else if`?

File: app/components/curated-content-item.js [hunk 2]
```diff
@@ -37,21 +37,28 @@ export default Component.extend(
...
 				// Sections use the same icons as categories
-				iconType = 'category';
+				iconType = 'grid';
 			} else {
 				// Default icon
 				iconType = 'article';
```
Comment (rybmat) [inline app/components/curated-content-item.js]: if you apply my comment above, then this whole if can be replaced with `iconType = typesWithDedicatedIcon[type] || 'article';`

File: app/templates/components/article-comments.hbs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div role="button" {{action 'toggleComments'}} class='show-comments-btn {{if isCollapsed 'collapsed'}}'>
-	{{i18n 'article.comments-label' count=commentsCount}}
-	{{svg 'chevron' viewBox='0 0 12 7' class='icon chevron'}}
+	<span>{{i18n 'article.comments-label' count=commentsCount}}</span>
+	{{svg 'wds-icons-menu-control' viewBox='0 0 12 12' class='wds-icon wds-icon-tiny'}}
```
Comment (rybmat) [inline app/templates/components/article-comments.hbs]: do we need to set viewBox here? We should use 'wds-icons-menu-control-tiny'

File: app/templates/components/article-media-thumbnail.hbs [hunk 4]
```diff
@@ -1,15 +1,15 @@
 {{! This wrapper is required so the chevron is positioned correctly }}
 <div class="article-media-thumbnail-image-wrapper">
 	{{#if isOgg}}
-		<a href="{{href}}">{{svg 'wds-icons-download' viewBox='0 0 24 24' class='icon'}}</a>
+		<a href="{{href}}">{{svg 'wds-icons-download' viewBox='0 0 24 24' class='wds-icon'}}</a>
```
Comment (rybmat) [inline app/templates/components/article-media-thumbnail.hbs]: do we need viewBox here? regular icons in wds are 24x24px

File: app/templates/components/article-media-thumbnail.hbs [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 	{{else}}
 		<a href="{{href}}">
 			<img alt="{{title}}"
 				 src="{{thumbnailUrl}}"
 				 height="{{if forcedHeight forcedHeight height}}"
 				 width="{{if forcedWidth forcedWidth width}}">
 			{{#if isLinkedByUser}}
-				{{svg 'chevron-linked' viewBox='0 0 20 20' class='chevron'}}
+				{{svg 'wds-icons-menu-control' viewBox='0 0 12 12' class='wds-icon wds-icon-tiny wds-icons-tiny chevron'}}
```
Comment (rybmat) [inline app/templates/components/article-media-thumbnail.hbs]: wds-icons-menu-control-tiny and remove viewBox

Comment (rybmat) [reply app/templates/components/article-media-thumbnail.hbs]: what is the difference between css classes `wds-icon-tiny` and `wds-icons-tiny`? is the later one defined at all?

File: app/templates/components/article-media-thumbnail.hbs [hunk 6]
```diff
@@ -19,6 +19,8 @@
 			{{svg 'wds-player-icon-play' class='wds-player-icon-play-medium'}}
 		</div>
 	{{/if}}
+
+	{{svg 'wds-icons-image' viewBox='0 0 24 24' role='img' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/article-media-thumbnail.hbs]: viewBox is not needed I think

File: app/templates/components/article-wrapper.hbs [hunk 7]
```diff
@@ -82,8 +82,8 @@
 			{{#each model.categories as |category|}}
 				<li>
 					<a href="{{unbound category.url}}" title="{{unbound category.title}}">
-						{{unbound category.title}}
-						{{svg 'chevron' viewBox='0 0 12 7' class='icon chevron'}}
+						<span>{{unbound category.title}}</span>
+						{{svg 'wds-icons-menu-control-tiny' viewBox='0 0 12 12' class='wds-icon wds-icon-tiny'}}
```
Comment (rybmat) [inline app/templates/components/article-wrapper.hbs]: viewBox is not needed IMO

File: app/templates/components/collapsible-menu.hbs [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <div role="button" {{action 'toggleMenu'}} class="{{if isCollapsed 'collapsed'}} title">
-	{{svg 'chevron' viewBox='0 0 12 7' class='icon chevron'}}
-	{{i18n tLabel count=tCount}}
+	<span>{{i18n tLabel count=tCount}}</span>
+	{{svg 'wds-icons-menu-control-tiny' viewBox='0 0 12 12' class='wds-icon wds-icon-tiny chevron'}}
```
Comment (rybmat) [inline app/templates/components/collapsible-menu.hbs]: viewBox is not needed IMO

File: app/templates/components/curated-content-item.hbs [hunk 9]
```diff
@@ -1,5 +1,6 @@
 <figure class="item-figure">
 	<div class="item-image" style={{style}}>
+		{{svg 'wds-icons-image' viewBox='0 0 24 24' role='img' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/curated-content-item.hbs]: viewBox is not needed IMO

File: app/templates/components/curated-content-section.hbs [hunk 10]
```diff
@@ -2,7 +2,7 @@
 	{{#if model.label}}
 		<div class="curated-content-section__header">
 			<a href="#" class="curated-content-section__back" {{action closeSection}}>
-				{{svg 'wds-icons-arrow' viewBox='0 0 20 16' role='img' class='icon'}}
+				{{svg 'wds-icons-arrow' viewBox='0 0 24 24' role='img' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/curated-content-section.hbs]: viewBox is not needed IMO

File: app/templates/components/fandom-app-smart-banner.hbs [hunk 11]
```diff
@@ -1,4 +1,6 @@
-<div class="fandom-app-smart-banner__close" role="button" {{action 'close'}}>{{svg 'close' viewBox='0 0 18 18'}}</div>
+<div class="fandom-app-smart-banner__close" role="button" {{action 'close'}}>
+	{{svg 'wds-icons-cross' viewBox='0 0 18 18' class='wds-icon wds-icon-small'}}
```
Comment (rybmat) [inline app/templates/components/fandom-app-smart-banner.hbs]: I don't think we need viewBox here

File: app/templates/components/featured-content-item.hbs [hunk 12]
```diff
@@ -1,4 +1,5 @@
 <figure>
+	{{svg 'wds-icons-image' viewBox='0 0 24 24' role='img' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/featured-content-item.hbs]: I don't think we need viewBox here

File: app/templates/components/featured-content.hbs [hunk 13]
```diff
@@ -1,10 +1,10 @@
 {{#if showChevrons}}
-	{{svg 'chevron-linked' viewBox='0 0 5 20' class='chevron chevron-left'}}
-	{{svg 'chevron-linked' viewBox='0 0 5 20' class='chevron chevron-right'}}
+	{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='wds-icon left'}}
```
Comment (rybmat) [inline app/templates/components/featured-content.hbs]: I don't think we need viewBox here

File: app/templates/components/featured-content.hbs [hunk 14]
```diff
@@ -1,10 +1,10 @@
...
+	{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='wds-icon right'}}
```
Comment (rybmat) [inline app/templates/components/featured-content.hbs]: I don't think we need viewBox here

File: app/templates/components/lightbox-wrapper.hbs [hunk 15]
```diff
@@ -4,7 +4,7 @@
 			{{#if closeButtonHidden}}
 				{{lightboxCloseButtonDelay}}
 			{{else}}
-				{{svg 'close' role='img' class='lightbox-close-button'}}
+				{{svg 'wds-icons-cross' role='img' viewBox='0 0 12 12' class='wds-icon wds-icon-tiny lightbox-close-button'}}
```
Comment (rybmat) [inline app/templates/components/lightbox-wrapper.hbs]: wds-icons-cross-tiny and get rid of viewBox

File: app/templates/components/portable-infobox-image-collection.hbs [hunk 16]
```diff
@@ -20,12 +20,12 @@
 <div class="image-collection-actions">
 	{{#if hasPreviousImage}}
 		<button class="action-previous" {{action 'switchImage' -1}}>
-			{{svg 'chevron-linked' viewBox='0 0 12 7'}}
+			{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/portable-infobox-image-collection.hbs]: I don't think we need viewBox here

File: app/templates/components/portable-infobox-image-collection.hbs [hunk 17]
```diff
@@ -20,12 +20,12 @@
...
 		</button>
 	{{/if}}
 	{{#if hasNextImage}}
 		<button class="action-next" {{action 'switchImage' 1}}>
-			{{svg 'chevron-linked' viewBox='0 0 12 7'}}
+			{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/portable-infobox-image-collection.hbs]: I don't think we need viewBox here

File: app/templates/components/trending-articles-item.hbs [hunk 18]
```diff
@@ -1,9 +1,10 @@
 <figure>
-	<div class="item-image" style={{imageStyle}}>
+	<div class="item-image">
+		{{svg 'wds-icons-image' viewBox='0 0 24 24' role='img' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/trending-articles-item.hbs]: I don't think we need viewBox here

File: app/templates/components/user-avatar.hbs [hunk 19]
```diff
@@ -8,7 +8,7 @@
 			<img src="{{avatarUrl}}" title="{{displayName}}" alt="{{displayName}}">
 		{{/if}}
 	{{else}}
-		{{svg 'avatar' viewBox='0 0 40 40' class='icon'}}
+		{{svg 'wds-avatar-icon' viewBox='0 0 40 40' class='wds-icon'}}
```
Comment (rybmat) [inline app/templates/components/user-avatar.hbs:11]: is this viewBox needed?

File: app/templates/components/widget-discussions-post.hbs [hunk 20]
```diff
@@ -41,23 +41,23 @@
 	</span>
 	<span class="post-counters">
 		<span class="upvote-count">
-			{{svg 'upvote' viewBox='0 0 24 24' class='icon upvote'}}
+			{{svg 'wds-icons-upvote' viewBox='0 0 24 24' class='wds-icon wds-icon-tiny upvote'}}
```
Comment (rybmat) [inline app/templates/components/widget-discussions-post.hbs]: why do we use regular size icon and apply wds-icon-tiny class to it? And do we need this viewbox?

File: app/templates/components/widget-discussions-post.hbs [hunk 21]
```diff
@@ -41,23 +41,23 @@
...
 			{{post.upvoteCount}}
 		</span>
 		<span class="reply-count">
-			{{svg 'reply' viewBox='0 0 24 24' class='icon replies'}}
+			{{svg 'wds-icons-reply' viewBox='0 0 24 24' class='wds-icon wds-icon-tiny replies'}}
```
Comment (rybmat) [inline app/templates/components/widget-discussions-post.hbs]: why do we use regular size icon and apply wds-icon-tiny class to it? And do we need this viewbox?

File: app/templates/components/widget-discussions-post.hbs [hunk 22]
```diff
@@ -41,23 +41,23 @@
...
 			{{post.repliesCount}}
 		</span>
 	</span>
 </div>
 <ul class="post-actions">
 	<li class="upvote-area upvote {{if post.userData.hasUpvoted 'has-upvoted'}}" role="button" {{action upvote post}}>
-		{{svg 'upvote' viewBox='0 0 24 24' class='icon upvote'}}
+		{{svg 'wds-icons-upvote' viewBox='0 0 24 24' class='wds-icon wds-icon-small upvote'}}
```
Comment (rybmat) [inline app/templates/components/widget-discussions-post.hbs]: why do we use regular size icon and apply wds-icon-small class to it? And do we need this viewbox?

File: app/templates/components/widget-discussions-post.hbs [hunk 23]
```diff
@@ -41,23 +41,23 @@
...
 		{{i18n 'widget-discussions.post-actions-upvote-label'}}
 	</li>
 	<li class="replies replies-area">
 		<a href="/d/p/{{post.threadId}}">
-			{{svg 'reply' viewBox='0 0 24 24' class='icon repliess'}}
+			{{svg 'wds-icons-reply' viewBox='0 0 24 24' class='wds-icon wds-icon-small replies'}}
```
Comment (rybmat) [inline app/templates/components/widget-discussions-post.hbs]: why do we use regular size icon and apply wds-icon-small class to it? And do we need this viewbox?

File: app/templates/components/widget-discussions.hbs [hunk 24]
```diff
@@ -5,7 +5,7 @@
 		</h2>
 		<a href="/d/f" class='heading-viewall'>
 			{{i18n 'widget-discussions.all-discussions-link-mobile'}}
-			{{svg 'arrow-right-long'}}
+			{{svg 'wds-icons-arrow' class='wds-icon wds-icon-tiny'}}
```
Comment (rybmat) [inline app/templates/components/widget-discussions.hbs]: why do we use regular size icon and apply wds-icon-tiny class to it?

File: app/templates/components/wikia-nav.hbs [hunk 25]
```diff
@@ -9,21 +9,22 @@
 					{{/if}}
 					<img src={{currentUser.avatarPath}} alt="{{currentUser.name}}">
 				</figure>
-				{{currentUser.name}}
-				{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='icon wds-menu-chevron'}}
+				<span>{{currentUser.name}}</span>
+				{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='wds-icon wds-menu-chevron'}}
```
Comment (rybmat) [inline app/templates/components/wikia-nav.hbs]: IMO we don't need viewBox here

File: app/templates/components/wikia-search.hbs [hunk 26]
```diff
@@ -3,7 +3,7 @@
 		<div class="wikia-search__container">
 			{{wikia-ui-components/icon-button
 				class='wikia-search__search-icon'
-				icon='search'
+				icon='wds-icons-magnifying-glass'
```
Comment (rybmat) [inline app/templates/components/wikia-search.hbs]: there is also wds-icons-magnifying-glass-small which is 18x18px

File: app/templates/components/wikia-ui-components/wiki-page-header.hbs [hunk 27]
```diff
@@ -8,7 +8,7 @@
 				{{#link-to 'wiki-page' mainPageTitle title=siteName class='wiki-page-header__site-link'}}
 					<span class="wiki-page-header__site-name">{{siteName}}</span>
 					{{i18n 'app.header-in-wiki-name-label-suffix'}}
-					{{svg 'arrow-right-long'}}
+					{{svg 'wds-icons-arrow' class='wds-icon wds-icon-tiny'}}
```
Comment (rybmat) [inline app/templates/components/wikia-ui-components/wiki-page-header.hbs]: if this should be tiny icon then use wds-icons-arrow-tiny, otherwise why we apply wds-icon-tiny class?

File: app/templates/components/wikia-user-profile.hbs [hunk 28]
```diff
@@ -2,7 +2,7 @@
 	<div class="wikia-nav__header">
 		<a role="button" {{action 'getBack' preventDefault=false}}
 				class="wikia-nav__back wikia-user-profile__back-link">
-			{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='icon wds-menu-chevron'}}
+			{{svg 'wds-icons-menu-control' viewBox='0 0 24 24' class='icon'}}
```
Comment (rybmat) [inline app/templates/components/wikia-user-profile.hbs]: viewbox is not needed IMO, and don't we need `wds-icon` class here?

**Review by rybmat — CHANGES_REQUESTED** [review]

File: package.json [hunk 29]
```diff
@@ -36,7 +36,8 @@
     "compression": "1.7.1",
     "cors": "2.8.4",
     "deep-extend": "0.5.0",
-    "design-system": "github:wikia/design-system#8.13.0",
+    "design-system": "github:wikia/design-system#9.0.1",
+    "ember-cli-concat": "1.3.0",
```
Comment (rybmat) [inline package.json]: why is this added in this PR?

File: app/styles/component/_featured-content.scss [hunk 30]
```diff
@@ -2,38 +2,54 @@
...
 
-	.chevron {
+	> .wds-icon {
 		fill: $white;
 		height: 100%;
 		position: absolute;
 		stroke: rgba(0, 0, 0, .5);
 		stroke-width: .25;
 		top: 0;
+		z-index: 1;
```
Comment (bkoval) [inline app/styles/component/_featured-content.scss]: How about using a $z-1 or defining a $z-index-featured-content-icon in wikia-variables.scss?

File: app/styles/mixin/image-placeholder.scss [hunk 31]
```diff
@@ -1,10 +1,8 @@
-// placeholder for setting up an image placeholder background
 %image-placeholder {
-	background: {
-		color: $gainsboro;
-		image: $brackets-svg-image;
-		position: center;
-		repeat: no-repeat;
-		size: contain;
-	}
+	color: $color-white;
```
Comment (bkoval) [inline app/styles/mixin/image-placeholder.scss]: So, sometimes we're using $color-white and sometimes #fff. Let's use one way everywhere :)

**Review by bkoval — CHANGES_REQUESTED** [review]

File: app/styles/mixin/image-placeholder.scss (hunk 31 above)
Comment (rybmat) [reply app/styles/mixin/image-placeholder.scss]: shouldn't we use `$wds-color-white` everywhere?

File: app/styles/component/_wikia-search.scss [hunk 32]
```diff
@@ -28,6 +28,11 @@ $border-bottom-height: 1px;
 
 	.wikia-search__clear {
 		width: $clear-icon-width;
+
+		svg {
+			fill: #656e78;
```
Comment (rybmat) [inline app/styles/component/_wikia-search.scss]: don't we have defined such colour somewhere? DS?

File: app/styles/module/_ads.scss [hunk 33]
```diff
@@ -105,6 +105,7 @@ $closeButtonHeight: 40px;
 	z-index: $z-modal-dialog;
 
 	.close {
+		color: #fff;
```
Comment (rybmat) [inline app/styles/module/_ads.scss]: $wds-color-white

File: app/styles/module/comments/index.scss [hunk 34]
```diff
@@ -1,19 +1,15 @@
 .show-comments-btn {
+	align-items: center;
 	background: none;
 	border-top: 1px solid $secondary-color;
 	color: #333;
```
Comment (rybmat) [inline app/styles/module/comments/index.scss]: `$wds-color-light-c6`

File: app/styles/component/_wikia-user-profile.scss [hunk 35]
```diff
@@ -19,8 +19,7 @@
 		line-height: 20px;
 
 		svg {
-			float: left;
-			margin-top: 5px;
+			fill: $color-blue-gray;
```
Comment (rybmat) [inline app/styles/component/_wikia-user-profile.scss]: wds-color-blue-gray

File: app/styles/component/_wikia-search.scss (hunk 32 above)
Comment (rybmat) [reply app/styles/component/_wikia-search.scss]: wds-color-slate-gray

File: app/styles/component/_site-head.scss [hunk 36]
```diff
@@ -113,20 +113,16 @@ body:not(.no-scroll) {
 	.site-head-icon {
 		position: relative;
 
-		.icon-button-icon {
-			fill: $color-blue-dark;
+		svg {
+			fill: #092344;
```
Comment (rybmat) [inline app/styles/component/_site-head.scss]: $wds-color-dark-blue

File: app/styles/component/_nav-menu.scss [hunk 37]
```diff
@@ -53,10 +50,18 @@ $nav-padding-horizontal: 6px;
...
-			padding-right: $nav-chevron-padding;
+			> span {
+				flex: 1;
+				overflow: hidden;
+				text-overflow: ellipsis;
+				white-space: nowrap;
+			}
+
+			> .wds-icon {
+				fill: #656e78;
```
Comment (rybmat) [inline app/styles/component/_nav-menu.scss]: $wds-color-slate-gray

File: app/styles/component/_featured-content.scss [hunk 38]
```diff
@@ -60,12 +76,29 @@
...
+			> * {
+				border: 1px solid #ccc;
+				border-radius: 4px;
+				display: block;
+				height: 9px;
+				width: 9px;
+			}
+
+			&.current > * {
+				background-color: $white;
```
Comment (rybmat) [inline app/styles/component/_featured-content.scss]: wds-color-white here and one line below

File: app/styles/component/_article-media-thumbnail.scss [hunk 39]
```diff
@@ -45,13 +40,15 @@
 	}
 
 	.chevron {
-		background-color: #fff;
+		background: #fff;
```
Comment (rybmat) [inline app/styles/component/_article-media-thumbnail.scss]: wds-color-white

**Review by rybmat — CHANGES_REQUESTED** [review]

File: app/styles/component/_featured-content.scss [hunk 40]
```diff
@@ -2,38 +2,54 @@
...
 			repeat: no-repeat;
 			size: contain;
 		}
 
+		img {
+			position: relative;
+		}
+
+		svg {
+			color: $color-white;
```
Comment (bkoval) [inline app/styles/component/_featured-content.scss]: So, which one is it? #fff or $color-white? 👯

**Review by rybmat — APPROVED** [review]

**Review by bkoval — APPROVED** [review]
